src/resources/rmd/hooks.R

```diff
@@ -44,6 +44,15 @@ knitr_hooks <- function(format, resourceDir, handledLanguages) {
         options[["results"]] <- "hold"
       }
     }
+    # handle specifically format which sets echo: false like revealjs
+    if (knitr::pandoc_to("revealjs", exact = TRUE)) {
+      # to adapt to true for engine that have no output
+      if (identical(options[["echo"]], FALSE) &&
+          options[["engine"]] %in% c("embed", "verbatim")
+      ) {
+        options[["echo"]] <- TRUE
+      }
+    }
```

Comment:
This is the proposal to have `echo: true` in revealjs by default for some engines. **revealjs** defaults to `echo: true` so using those engine requires to know that setting `echo: true` on chunk is needed. 

````markdown

```{embed}
#| echo: true
#| file: code.R
```
````

Otherwise, it does not show as explained in the issue. 

Maybe it is too much... would like your thought on that ? 
Other solution would be to document this (maybe in https://quarto.org/docs/presentations/revealjs/#code-echo)

Reply:
If it never makes sense to have `echo = FALSE` for embed then we might just want to set it unqualified? My concern is that there are other formats which set echo = FALSE by default (all the presentation formats, and certainly any dashboard formats we create in the future) so hard-coding to "revealjs" is probably not the thing to do.

Reply:
> that there are other formats which set echo = FALSE by default

I thought we would add them in the list of exception, but I thought we add only revealjs currently. 

> If it never makes sense to have echo = FALSE for embed then we might just want to set it unqualified? 

What do you mean by unqualified ? Not sure to grasp what that entails...

For more context, 

Those formats are all derived from the `cat` engine and are engine with no output but just code part (related to `knitr::engine_output()`). So `echo: FALSE` means nothing will be shown at all, and you probably don't want that, even if you want to hide all the other source code for chunks.

We did not make `echo = TRUE` by default for those engines in this **knitr** because `echo` can be used to filter what to show depending on format (`echo = knitr::is_latex_output()` for example). With Quarto, using expression for `echo` is not what we recommend, so it seems not a problem to do that.

Reply:
I was thinking that for those engines running in quarto we just force `echo = TRUE`. The problem with maintaining a list is that it doesn't cover custom formats (i.e. we can't actually know the list).